sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/InternalRow.scala

```diff
@@ -119,4 +119,26 @@ object InternalRow {
     case v: MapData => v.copy()
     case _ => value
   }
+
+  /**
+   * Returns an accessor for an InternalRow with given data type and ordinal.
+   */
+  def getAccessor(dataType: DataType, ordinal: Int): (InternalRow) => Any = dataType match {
```

Comment:
Maybe we can make the accessor accept `SpecializedGetters`, so this can be reused in #20984 too...

Reply:
Ah, no. The accessor in #20984 takes ordinal as input parameter.

Reply:
Well you could generalize to make it take an ordinal?

Reply:
Ok.